include/seqan/graph_types/graph_impl_directed.h

```diff
@@ -74,55 +74,112 @@ class Graph<Directed<TCargo, TSpec> >
 
 //____________________________________________________________________________
 
+        Graph() {}
 
-        Graph() {
-            SEQAN_CHECKPOINT
+        ~Graph()
+        {
+            clear(*this);
+        }
+
+        Graph(Graph const & _other) :
+            data_allocator(_other.data_allocator)
+        {
+            _copyGraph(*this, _other);
         }
 
-        ~Graph() {
-            SEQAN_CHECKPOINT
+        Graph const & operator = (Graph const & _other)
+        {
+            if (this == &_other)
+                return *this;
+
+            clear(*this);
+            data_allocator = _other.data_allocator;
+            _copyGraph(*this, _other);
+            return *this;
+        }
+};
+
+template <typename TCargo>
+class Graph<Directed<TCargo, WithSourceId> >
+{
+    public:
+        typedef typename VertexIdHandler<Graph>::Type TVertexIdManager_;
+        typedef typename EdgeIdHandler<Graph>::Type TEdgeIdManager_;
+        typedef typename EdgeType<Graph>::Type TEdgeStump_;
+        typedef Allocator<SinglePool<sizeof(TEdgeStump_)> > TAllocator_;
+
+        String<TEdgeStump_*> data_vertex;            // Pointers to EdgeStump lists
+        String<TEdgeStump_*> data_vertex_in;         
+        TVertexIdManager_ data_id_managerV;
+        TEdgeIdManager_ data_id_managerE;
+        TAllocator_ data_allocator;
+
+//____________________________________________________________________________
+
+        Graph() {}
+
+        ~Graph()
+        {
             clear(*this);
         }
 
         Graph(Graph const & _other) :
             data_allocator(_other.data_allocator)
         {
-            SEQAN_CHECKPOINT
-            _copyGraph(_other, *this);
+            _copyGraph(*this, _other);
         }
 
-        Graph const& operator = (Graph const & _other) {
-            SEQAN_CHECKPOINT
-            if (this == &_other) return *this;
+        Graph const & operator = (Graph const & _other)
+        {
+            if (this == &_other)
+                return *this;
+
             clear(*this);
             data_allocator = _other.data_allocator;
-            _copyGraph(_other, *this);
+            _copyGraph(*this, _other);
             return *this;
         }
 };
 
-
 //////////////////////////////////////////////////////////////////////////////
 // INTERNAL FUNCTIONS
 //////////////////////////////////////////////////////////////////////////////
 
 //////////////////////////////////////////////////////////////////////////////
 
-template<typename TCargo, typename TSpec>
-inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*>&
-_getVertexString(Graph<Directed<TCargo, TSpec> > const& g) {
-    SEQAN_CHECKPOINT
+template <typename TCargo, typename TSpec>
+inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*> &
+_getVertexString(Graph<Directed<TCargo, TSpec> > const & g)
```

Comment:
I decided to leave this function like it was before (except some formatting and remove SEQAN_CHECKOUT) and remove the one I create one for the in vertices. _copyGraph no longer uses this function but it is still used in the test suite.